tidysic/parser/tree.py

```diff
@@ -25,8 +32,46 @@ def _parse(self) -> None:
         """
         for path in self._root.iterdir():
             if path.is_dir():
-                self.children.add(Tree(path))
+                child = Tree(path)
+                if child.common_tags is not None:
+                    self.children.add(child)
+                else:
+                    self.clutter_files.add(TaggedFile(path))
             elif AudioFile.is_audio_file(path):
                 self.audio_files.add(AudioFile(path))
             else:
-                self.clutter_files.add(path)
+                self.clutter_files.add(TaggedFile(path))
+
+        self._tag_clutter()
+
+    def _tag_clutter(self):
+        """
+        Tags non-audio files with the tags common to all audio files in the same
+        directory and subdirectories.
+        """
+        self._find_common_tags()
+        self._apply_common_tags_to_clutter()
+
+    def _find_common_tags(self):
+        """
+        Finds the common tags shared by the given tagged objects.
+        """
+        children_tags = [
+            child.common_tags
+            for child in self.children
+            if child.common_tags is not None
+        ]
+        all_tags = list(chain(self.audio_files, children_tags))
```

Comment:
`all_tags` could be `self.audio_files.union(children_tags)` if the previous comment is applied.

Reply:
After reflection, our sets are build from the hash that is computed from the `path` attribute. Two children with the same `common_tag` would still be considered different because of their `path`. A set doesn't bring anything here.